pytest.ini

```diff
@@ -13,3 +13,5 @@ testpaths =
 junit_logging_reruns = all
 filterwarnings =
     ignore:Module already imported so cannot be rewritten.*hypothesis:pytest.PytestAssertRewriteWarning
+
+strict_xfail = True
```

Comment:
I mentioned we can merge once ci is green, but actually I'd like to know if this line is strictly necessary.

Reply:
TL;DR: not strictly necessary, is a technical implement, is safe to remove

With this, xpass (a test is marked as xfail but passes) is reported as a failure. Is useful if the test suite has ~50 unexpected xpasses---with this, they are all reported at the bottom of the log; otherwise, need to fish out them manually.

Also, this is only active if run via `pytest path/to/test/file` and not if run via pytorch-y `python path/to/test/file`.

Reply:
SGTM, but tagging @ezyang for confirmation

Reply:
`strict_xfail` isn't a thing. Did you mean `xfail_strict`?

Reply:
The defaults change is fine. I actually thought pytest defaulted this way. Note that some devs do use pytest for local dev, it is a supported flow.